stbt_auto_selftest.py

```diff
@@ -92,23 +96,33 @@ def main(argv):
     os.chdir(root)
 
     if cmdline_args.command == 'generate':
-        generate()
+        generate(cmdline_args.source_files)
     elif cmdline_args.command == 'validate':
         return validate()
     else:
         assert False
 
 
-def generate():
-    tmpdir = generate_into_tmpdir()
+def generate(source_files):
+    valid_source_files = _valid_source_files(source_files)
```

Comment:
Good idea. I've added a check that the file exists. I still need to error for source files that have invalid python syntax, or that don't produce an auto-selftest file. For those cases I will need to do some bigger refactorings.